karapace/protobuf/proto_normalizations.py

```diff
@@ -0,0 +1,282 @@
+"""
+Copyright (c) 2024 Aiven Ltd
+See LICENSE for details
+"""
+from karapace.dependency import Dependency
+from karapace.protobuf.enum_constant_element import EnumConstantElement
+from karapace.protobuf.enum_element import EnumElement
+from karapace.protobuf.extend_element import ExtendElement
+from karapace.protobuf.field_element import FieldElement
+from karapace.protobuf.group_element import GroupElement
+from karapace.protobuf.message_element import MessageElement
+from karapace.protobuf.one_of_element import OneOfElement
+from karapace.protobuf.option_element import OptionElement
+from karapace.protobuf.proto_file_element import ProtoFileElement
+from karapace.protobuf.rpc_element import RpcElement
+from karapace.protobuf.schema import ProtobufSchema
+from karapace.protobuf.service_element import ServiceElement
+from karapace.protobuf.type_element import TypeElement
+from karapace.schema_references import Reference
+from typing import List, Mapping, Optional, Sequence
+
+
+def sort_by_name(element: OptionElement) -> str:
+    return element.name
+
+
+class NormalizedRpcElement(RpcElement):
+    pass
+
+
+class NormalizedServiceElement(ServiceElement):
+    rpcs: Optional[Sequence[NormalizedRpcElement]] = None
+
+
+class NormalizedFieldElement(FieldElement):
+    pass
+
+
+class NormalizedExtendElement(ExtendElement):
+    fields: Optional[Sequence[NormalizedFieldElement]] = None
+
+
+class NormalizedTypeElement(TypeElement):
+    nested_types: Sequence["NormalizedTypeElement"]
+
+    def to_schema(self) -> str:
+        """Convert the object to valid protobuf syntax.
+
+        This must be implemented by subclasses.
+        """
+        raise NotImplementedError()
```

Comment:
```suggestion
```

issue: This is identical to inherited (non-) implementation?

Reply:
pylint was complaining that wasn’t implemented, how can I get rid of that problem without replicating the non implementation? With an @abstract annotation?

Reply:
Ah, yes it might be happy with a `@abstractmethod` (`abc.abstractmethod`), it's appropriate but I don't know whether pylint understands it.

Reply:
Oh wait, what you'd need is to add `abc.ABC` as base to the class.